packages/viewer/src/modules/drawing/DrawingModule.vue

```diff
@@ -107,94 +158,95 @@ watch(hasLoaded, () => {
         addKmlToDrawing()
     }
 })
-// watching feature IDs so that we can react whenever one is removed through the "trash button"
-watch(featureIds, (next, last) => {
+
+// Watching feature IDs so we can react whenever one is removed through the "trash button"
+watch(featureIds, (next: string[], last: string[]) => {
     const removed = last.filter((id) => !next.includes(id))
     if (removed.length > 0) {
         log.debug(`${removed.length} feature(s) have been removed, removing them from source`)
         const source = drawingLayer.getSource()
+        if (!source) return
         source
             .getFeatures()
-            .filter((feature) => removed.includes(feature.getId()))
+            .filter((feature) => removed.includes(String(feature.get('id'))))
             .forEach((feature) => source.removeFeature(feature))
-        debounceSaveDrawing()
+        debounceSaveDrawing().catch((error) => {
+            log.error('Error while saving drawing after feature removal', error)
+        })
     }
 })
+
+// Workaround for legacy drawings and icon set mapping
 watch(availableIconSets, () => {
-    // Here this is a workaround for the legacy drawing opened in admin mode. In this case if
-    // the availableIconSets is not yet loaded while parsing the KML, we cannot build the correct
-    // DrawingIcon for default set icon because the icon name in the legacy icon service (mf-chsdi3)
-    // did not have any numbered prefix. This means that without this workaround the icon preselection
-    // from the set will not work and when modifying the drawing you might end up with a brocken
-    // drawing.
-    log.debug(
-        'New iconsets available update all drawing features',
-        drawingLayer.getSource().getFeatures()
-    )
+    const source = drawingLayer.getSource()
+    if (!source) return
```

Comment:
Same here, otherwise it's visually just one chunk of statements with a not so obvious `return` :smile: